precompiles/ics20/types.go

```diff
@@ -323,3 +321,27 @@ func checkAllocationExists(allocations []transfertypes.Allocation, sourcePort, s
 
 	return spendLimit, 0, fmt.Errorf(ErrNoMatchingAllocation, sourcePort, sourceChannel, denom)
 }
+
+// convertToAllocation converts the transfer types Allocation to the ICS20 Allocation.
```

Comment:
I think a little more context as to what discerns the Allocation from ICS20 Allocation could help here IMO
```suggestion
// convertToAllocation converts the Allocation type from the IBC transfer types to our implementation of ICS20 Allocation. The conversion maps the native SDK coin type to the custom coin type, which uses Ethereum native big integers.
```